### PR TITLE
Build from source on Windows

### PR DESCRIPTION
**Pull request by jakirkham**
Fixes https://github.com/conda-forge/yasm-feedstock/issues/2

Attempts to build from source on Windows using `cmake`.

### CODE REVIEW

**Comment by jakirkham** [comment]
This seems to have gone a lot smoother than even expected. Though there is one disconcerting development. During [installation]( https://ci.appveyor.com/project/conda-forge/yasm-feedstock/build/1.0.14/job/nc47y30pw2p1bipp#L165 ) it appears the `vs2013_runtime` is pulled in at the same time as the `vs2008_runtime`. I'm guessing we need to use the `vc` package to fix this. Not sure though. Thoughts @msarahan?

Edit: NVM appears this is was a change made to our CMake package. ( https://github.com/conda-forge/cmake-feedstock/pull/27 )

File: recipe/bld.bat [hunk 1]
```diff
@@ -1 +1,19 @@
...
+if errorlevel 1 exit 1
+
+cd build
+if errorlevel 1 exit 1
+
+cmake -G "NMake Makefiles" ^
+         -DCMAKE_BUILD_TYPE:STRING=RELEASE ^
+         -DCMAKE_PREFIX_PATH=%LIBRARY_PREFIX% ^
+         -DCMAKE_INSTALL_PREFIX:PATH=%LIBRARY_PREFIX% ^
+         -DBUILD_STATIC_LIBS:BOOL=ON ^
```
Comment (jakirkham) [inline recipe/bld.bat:11]: We are only building static libraries on Unix. So do the same thing on Windows.

**Comment by jakirkham** [comment]
Could you please review this, @patricksnape?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -5,30 +5,36 @@ package:
...
-  url: http://www.tortall.net/projects/yasm/releases/yasm-{{ version }}-win64.exe  # [win64]
-  sha1: f7872c6db3b4d93fd5a51717364d438108473149                                   # [win64]
+  fn: yasm-{{ version }}.tar.gz
+  url: http://www.tortall.net/projects/yasm/releases/yasm-{{ version }}.tar.gz
+  sha1: b7574e9f0826bedef975d64d3825f75fbaeef55e
 
 build:
   number: 0
-  skip: true  # [unix and py3k]
+  skip: true  # [(unix and py3k) or (win and py36)]
```
Comment (patricksnape) [inline recipe/meta.yaml:14]: Why skip Python 3.6?

Comment (jakirkham) [reply recipe/meta.yaml:14]: Python 3.5 and 3.6 use the same VC runtime. So I just picked one and skipped the other.

**Comment by jakirkham** [comment]
Thanks @patricksnape.

**Pull request merged**